### PR TITLE
Support for ECS 2 AMI

### PR DESCRIPTION
**Pull request by fjromerom**
### Summary

- Provides support for the new ECS 2 AMI
- Removed broken and unnecessary echo within the try_set_instance_infodir() function.
- Using mode debug, it created an additional directory to write the output of "last" because it loaded twice the init() function. I removed it from the get_sysinfo() function, because first I don't know why this is included in this function since it's not system info per se and second I don't see any reason why we may need this information, I've never used it.

### Testing

- [X] Works properly on Amazon Linux
- [X] Works properly on Amazon Linux 2
- [X] Works properly on RHEL 7
- [X] Works properly on Debian 8
- [X] Works properly on Ubuntu 14.04

New tests cover the changes: yes

### Licensing

This contribution is under the terms of the Apache 2.0 License:

### CODE REVIEW

File: ecs-logs-collector.sh
Comment (jahkeup) [inline ecs-logs-collector.sh]: AL2 is systemd based, so we should gather the services with `systemctl`, like so:

```suggestion
      systemctl list-units > ${info_system}/services.txt 2>&1
```

File: ecs-logs-collector.sh
Comment (jahkeup) [inline ecs-logs-collector.sh]: Restart on AL2 should use `systemctl` to interact with the `docker.service`, like:

```suggestion
          systemctl restart docker.service
```

File: ecs-logs-collector.sh
Comment (jahkeup) [inline ecs-logs-collector.sh:561]: nit: can we drop this extra newline?

File: ecs-logs-collector.sh
Comment (jahkeup) [inline ecs-logs-collector.sh]: AL AMI will need to keep the upstart instructions to restart the Agent:
```suggestion
        stop ecs; start ecs
```

File: ecs-logs-collector.sh
Comment (jahkeup) [inline ecs-logs-collector.sh:591]: The code in the case statements are relatively similar in both the `amazon` and `amazon2` cases, can we combine them until the point they diverge (ie: the primary difference is `systemctl restart ecs.service` vs `stop ecs; start ecs`)?

Something like:

```bash
case "${os_name}" in
  amazon|amazon2)
    if [ -e /etc/ecs/ecs.config ] &&  grep -q "^\s*ECS_LOGLEVEL=debug" /etc/ecs/ecs.config
    then
      info "Debug mode is already enabled."
    else
      echo "ECS_LOGLEVEL=debug" >> /etc/ecs/ecs.config
    fi
    
    try "restart the Amazon ECS Container Agent to enable debug mode"
    case "${os_name}" in
      amazon) stop ecs; start ecs ;;
      amazon2) systemctl restart ecs.service ;;
    esac
    ;;
  *)
    warning "the current operating system is not supported."
    ;;
esac
```

Though we could apply this same pattern to the docker debug case (updated above), I think that we should leave it as there may be further updates and differences that are likely to be handled in the case of AL2 where ecs won't.

File: ecs-logs-collector.sh
Comment (jahkeup) [inline ecs-logs-collector.sh:346]: Let's quote the use of the `${dstdir}` here for expansion's sake and not call `journalctl` by absolute path, instead lets let it be anywhere on the path with plain ol `journalctl -u docker`.
```suggestion
        journalctl -u docker > "${dstdir}"/docker
```

**Review by jahkeup — CHANGES_REQUESTED** [review]
@fjromerom thanks for tackling the addition of AL2 to the log collector!

I like the overall approach, and have a few pieces of feedback and changes I think we should make before merging this.

File: ecs-logs-collector.sh
Comment (fjromerom) [reply ecs-logs-collector.sh]: Agree.

File: ecs-logs-collector.sh
Comment (fjromerom) [reply ecs-logs-collector.sh]: It also works with service but I agree.

File: ecs-logs-collector.sh
Comment (fjromerom) [reply ecs-logs-collector.sh:561]: I just wanted to keep the same structure in the case. We can agree a structure for next releases.

File: ecs-logs-collector.sh
Comment (fjromerom) [reply ecs-logs-collector.sh]: Yep, that's right.

File: ecs-logs-collector.sh
Comment (fjromerom) [reply ecs-logs-collector.sh:591]: I thought about that as well but I wanted to do as minor changes as possible to support AL2. I will combine them in the next update.

File: ecs-logs-collector.sh
Comment (fjromerom) [reply ecs-logs-collector.sh:346]: That sounds good for me. I just kept the same format as other Systemd systems, so if we change this we should change the others too. Also, every mention to `${dstdir}` is not quoted, I would leave these changes for the next update so they all are updated in the same commit.

File: ecs-logs-collector.sh
Comment (jahkeup) [reply ecs-logs-collector.sh:561]: I'm all for incremental improvements, I don't see a reason to not include such an improvement with this change. Please do if its practical to do so.

File: ecs-logs-collector.sh
Comment (jahkeup) [reply ecs-logs-collector.sh:346]: I'm all for incremental improvements, and though the precedent exists (much of this appears to be copied from there), I would prefer we take this opportunity to make this addition better than what came before :)

File: ecs-logs-collector.sh
Comment (jahkeup) [reply ecs-logs-collector.sh:591]: I don't see a good reason to not make this change given it cleans this up; that said we don't have to change it either. I'll leave it to your discretion.

**Review by jahkeup — APPROVED** [review]
This looks good to move forward with! There are a couple more changes that could be made, but aren't necessary to merge.

**Pull request merged**
